config/jobs/periodic/docker-in-docker/periodic-dind-build.yaml

```diff
@@ -0,0 +1,213 @@
+              then 
+                echo "DOCKER_VERS, CONTAINERD_VERS, PACKAGING_REF are in env.list" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              else 
+                echo "DOCKER_VERS, CONTAINERD_VERS and/or PACKAGING_REF are not in env.list" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+                exit 1
+              fi
+              set -o allexport
+              source env.list
+            else
+              echo "There is no env.list" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+            fi
+            
+            # generate the env-distrib.list
+            echo "*** * env-distrib.list * ***" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+            mkdir docker-ce-packaging
+            pushd docker-ce-packaging
+            git init
+            git remote add origin  https://github.com/docker/docker-ce-packaging.git
+            git fetch --depth 1 origin ${PACKAGING_REF}
+            git checkout FETCH_HEAD
+
+            make REF=${DOCKER_VERS} checkout
+            popd
+
+            # get the packages list in the env_distrib.list
+            echo DEBS=\"`cd docker-ce-packaging/deb && ls -1d debian-* ubuntu-*`\" > env-distrib.list
+            echo RPMS=\"`cd docker-ce-packaging/rpm && ls -1d centos-* fedora-*`\" >> env-distrib.list
+
+            rm -rf docker-ce-packaging
+
+            # check the env.list (versions of docker-ce, containerd and list of packages)
+            if [[ -f env-distrib.list ]]
+            # if there is env.list and env-distrib.list
+            then
+              cat env-distrib.list 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              # check if there are two variables in env-distrib.list
+              if grep -Fq "DEBS" env-distrib.list && grep -Fq "RPMS" env-distrib.list
+              then 
+                echo "DEBS and RPMS are in env-distrib.list" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              else 
+                echo "DEBS and/or RPMS are not in env-distrib.list" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+                exit 1
+              fi
+              source env-distrib.list
+            else
+              echo "There is no env-distrib.list" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              exit 1
+            fi
+
+            # build docker_ce and containerd
+            echo "*** ** BUILD ** ***" 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+            CONT_NAME=docker-build
+            docker run --env DOCKER_SECRET_AUTH --env PATH_SCRIPTS --env PATH_LOG_PROWJOB -d -v /workspace:/workspace --privileged --name $CONT_NAME ${PATH_IMAGE_BUILD} ${PATH_SCRIPTS}/build.sh 
+            docker logs -f $CONT_NAME
+            status_code="$(docker container wait $CONT_NAME)"
+            if [[ ${status_code} -ne 0 ]]
+            then
+              echo "The docker supposed to build the packages has failed." 2>&1 | tee -a ${PATH_LOG_PROWJOB}
+              exit 1
+            fi
+
+            # change the containerd environment variable if CONTAINERD_VERS=0
```

Comment:
Can you please explain on why `CONTAINERD_VERS` might be 0
why is the last version from COS bucket being re-tested if the new one isn't available?

Reply:
In case there is no new version of containerd, we would set CONTAINERD_VERS to 0, that way, we would avoid building and pushing it once again